linera-execution/src/evm/revm.rs

```diff
@@ -272,13 +273,28 @@ fn address_to_user_application_id(address: Address) -> ApplicationId {
     ApplicationId::new(CryptoHash::try_from(&vec as &[u8]).unwrap())
 }
 
-#[repr(u8)]
-#[derive(Debug, TryFromPrimitive)]
-enum PrecompileTag {
+/// Some functionalities from the BaseRuntime
+#[derive(Debug, Serialize, Deserialize)]
+enum BasePrecompileTag {
+    /// Key prefix for `chain_id`
+    ChainId,
+    /// Key prefix for `application_creator_chain_id`
+    ApplicationCreatorChainId,
+    /// Key prefix for `chain_ownership`
+    ChainOwnership,
+    /// Key prefix for `read_data_blob`
+    ReadDataBlob,
+    /// Key prefix for `assert_data_blob_exists`
+    AssertDataBlobExists,
+}
```

Comment:
This is still missing lots of methods. What's the plan for adding it?

Reply:
It is generally preferred to have smaller PRs.
Some functions are not going to be inserted like `fn application_parameters` because it contains the constructor parameters.
But yes, other functions could to be inserted:
* Maybe `read_system_timestamp` since it in micros vs second for EVM.
* The `read_chain_balance`.
* The `read_owner_balance` is normally available directly in EVM.
* The `read_owner_balances` and `read_balances_owner` could be done

The above is contingent on having correct understanding of balances in EVM.

Then the `http_request` requires more work and thinking for a good test case.

The events require their own work as well.

The `create_application` requires more work in relation to `create / create2`.